adapter/src/debug_session.rs

```diff
@@ -744,7 +744,11 @@ impl DebugSession {
             bail!(as_user_error(r#"Either "program" or "pid" is required for attach."#));
         }
 
-        self.target = Initialized(self.debugger.create_target("", None, None, false)?);
+        let program = match &args.program {
+            Some(program) => program,
+            None => "",
```

Comment:
I'd prefer to explicitly use `self.debugger.create_target` in this case, for clarity.

Reply:
Updated, have a look if it makes sense now?